xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/less/drawer.less

```diff
@@ -20,87 +20,133 @@
 @xwiki-drawer-menu-category-bg:          @xwiki-drawer-menu-item-seperator-color;
 @xwiki-drawer-menu-category-color:       lighten(@xwiki-drawer-menu-item-color, 10%);
 
-.drawer-overlay {
-  z-index: 5200;
-}
-
 .drawer--right .drawer-nav {
   right: -@xwiki-drawer-width;
 }
 
-.drawer-nav {
+
+
+#tmDrawer.drawer-nav {
   background-color: @xwiki-drawer-bg;
   width: @xwiki-drawer-width;
-  z-index: 5300;
+  z-index: 5200;
+  position: fixed;
+  top: 0;
+  overflow: hidden;
+  height: 100vh;
+  right: -@xwiki-drawer-width;
+  transition: right .6s cubic-bezier(0.190, 1.000, 0.220, 1.000), visibility .6s;
+  &.closed {
+    // We use visibility here to avoid breaking the animations.
+    visibility: hidden;
+  }
+  &.opened {
+    right: 0;
+  }
 
-  // Brand =================================================================
-  .drawer-brand {
-    background-color: @xwiki-drawer-brand-bg;
-    border-bottom: 1px solid @xwiki-drawer-brand-border;
-    font-weight: 400;
-    line-height: inherit;
-    padding: 20px 16px;
+  & + .drawer-overlay {
+    position: fixed;
+    display: none;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    opacity: 0.2;
+    background-color: #fff - @xwiki-page-content-bg;
```

Comment:
Usually I remove it / avoid it, but here I think it makes sense because:
* This is an overlay which is supposed to be transparent (on the opposite to most elements of XWiki that don't really 'hide' anything and therefore have no business prefering transparency over full colors)
* The content behind it is not clickable

The contrast is broken on the other elements, but I think it's an okay case where it's all inactive. Changing this would lead to difficult questions and I'm not sure there's a solution that is marginally better.

Making this overlay a full color is not a proper solution.
And AFAIK the only other solution here is to remove this overlay, but I think it's useful for understandability to have it when the drawer opens.